src/cmd/tools/doctest/doctest.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	operationRegEx  = `(?s)<operation>.*?</operation>|(?s)<operation_json>.*?</operation_json>`
+	validationRegEx = `(?s)<validation>.*?</validation>`
+
+	operationOpen      = `<operation>`
+	operationClose     = `</operation>`
+	operationJSONOpen  = `<operation_json>`
+	operationJSONClose = `</operation_json>`
+	validationOpen     = `<validation>`
+	validationClose    = `</validation>`
+
+	all = `<operation>|</operation>|<validation>|</validation>`
+)
+
+var (
+	file           string
+	destDir        string
+	markdown, bash bool
+
+	goPath = build.Default.GOPATH + "/src/"
+)
+
+func init() {
+	flag.StringVar(&file, "file", "", "File to be parsed")
+	flag.StringVar(&destDir, "dest", "", "Directory where the output will be saved")
```

Comment:
might be simpler to just take two strings - one for destination filename for bash and one for markdown. default to empty. if the value is not the empty string, then you generate the file, otherwise you don't.